addons/explosives/functions/fnc_getDetonators.sqf

```diff
@@ -1,25 +1,21 @@
 /*
- * Author: Garth 'L-H' de Wet
- * Returns all the detonators of the unit
+ * Author: Garth 'L-H' de Wet, mharis001
+ * Returns all detonators the given unit has.
  *
  * Arguments:
  * 0: Unit <OBJECT>
  *
  * Return Value:
- * Configs of all detonators <ARRAY>
+ * Config names of detonators <ARRAY>
  *
  * Example:
- * _detonators = [player] call ACE_Explosives_fnc_getDetonators;
+ * [_player] call ace_explosives_fnc_getDetonators
  *
  * Public: Yes
  */
 #include "script_component.hpp"
-// IGNORE_PRIVATE_WARNING(_detonators);
 
 params ["_unit"];
-TRACE_1("params",_unit);
+TRACE_1("Getting detonators",_unit);
 
-private _result = (items _unit) select {getNumber (ConfigFile >> "CfgWeapons" >> _x >> QGVAR(Detonator)) == 1};
-_result = _result arrayIntersect _result;
-
-_result
+(_unit call EFUNC(common,uniqueItems)) select {getNumber (configFile >> "CfgWeapons" >> _x >> QGVAR(Detonator)) == 1};
```

Comment:
here you can again move the `configFile >> "CfgWeapons"` outside the loop.
How often does this function run? That's a interaction menu condition right? so probably worth it.

Reply:
Keep forgetting to do that..